cmd/ig/main.go

```diff
@@ -38,7 +38,8 @@ func main() {
 		Use:   "ig",
 		Short: "Collection of gadgets for containers",
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			if os.Geteuid() != 0 {
+			// We allow ig version to not be run as root.
+			if os.Geteuid() != 0 && os.Args[1] != "version" {
```

Comment:
Should check for `len(os.Args) > 1` before accessing `os.Args[1]` here. Maybe you can use `args[0]` instead (also with a proper length check).